src/lib.rs

```diff
@@ -86,6 +86,98 @@ where
         self.read_register(Register::WHO_AM_I)
     }
 
+    /// Read `STATUS_REG` of sensor
+    pub fn status(&mut self) -> Result<Status, E> {
+        let sts = self.read_register(Register::STATUS_REG)?;
+        Ok(
+            Status{
```

Comment:
nit: could you refactor this code into a private `Status` constructor?

Reply:
Quick question, do you mean a simple `from_u8(from: u8)` method which implements the bit checking and conversion to `bool` or a normal `new(overrun: bool, ...)`?

Reply:
the former, `Status::from_u8`